### ISSUE TITLE
Feature request : Package of the day on homepage

### ISSUE THREAD

**Issue by Skyost** [dart-lang/pub-dev#3182]
Hi,    
Here's a little improvement idea I had for pub.dev. It would be a little section on the homepage that shows a random widget with something like 90+ score. It has some advantages :

1. You help developers getting their package discovered by the community.
2. You improve the creativity amongst developers (as they see a highlighted package, they can think "Hmm... I could build an app around this package !").
3. Personally, I'm regularly searching for new packages to add into my project (and maybe I'm not alone).

So it would be a win-win situation between you and the developers.

Thanks for reading !

**Comment by isoos** [comment]
@Skyost: What process do you follow when searching and adding packages to your project?

The landing page (https://pub.dev/) has a random list of Flutter favorite packages, and also a random list of top packages. The current process uses the top 100 (or 150) package as the source of randomization of the later, but of course we could increase the pool of such packages. 

What change would be useful for you on that page?

**Comment by Ephenodrom** [comment]
@isoos The big problem with the landing page is that it seems that you always see the same packages. I think this is more of a problem because the scoring stops at 100 points and the top packages are not changing at all. 

So a "package of the day" or a list of "new packages" would be nice.

Another idea would be to add "tags" to your package to make it easier to filter.
If i want to look for packages that have to do with the camera i could filter for the tag "camera".

**Comment by isoos** [comment]
> The big problem with the landing page is that it seems that you always see the same packages.

We have changed that last week, we choose randomly. Although if you are not logged in, you'll get the cached page, which changes only after 10 minutes (or an hour, I'm not sure). If you are logged it, all page request will get you a new set.

> If i want to look for packages that have to do with the camera i could filter for the tag "camera".

What's your experience with searching for "camera" as a search string? Is there a problem with that?

**Comment by Skyost** [comment]
@isoos

> What process do you follow when searching and adding packages to your project?

Usually I search them while I'm coding. I try to do something, and if it's too complicated / long / not good enough, I search for an existing flutter package.

> The landing page (https://pub.dev/) has a random list of Flutter favorite packages, and also a random list of top packages. The current process uses the top 100 (or 150) package as the source of randomization of the later, but of course we could increase the pool of such packages. What change would be useful for you on that page?

Just like @Ephenodrom, I find this list too restrictive as your plugin need to be embedded in the [Flutter Favorite program](https://flutter.dev/docs/development/packages-and-plugins/favorites) and thus you always see the same plugins on the front page (which are useful, I agree but "too basic" : those are the plugins you include first in your app). My suggestion was to highlight just one plugin per day (it can be based on criterias : a new / an old plugin, should be _x_+ score) that stands out from the rest.

**Comment by Ephenodrom** [comment]
@isoos 
> What's your experience with searching for "camera" as a search string? Is there a problem with that?

Yes this will work too. But for example i have a utils package that covers different problems.
* String operations
* Math operations
* X509 operations
* Domain operations

When i search for "domain" my package comes up on the first page, but some other packages are there too, that have nothing or less to do with domain operations. Therefore if a package covers different problems, i think tagging it with different words would make sense.

This could work like on github where you can add a certain amount of tags to your github repository.

**Comment by sigurdm** [comment]
I think we are doing something like this at this point. The homepage is showing a random selection of top packages in different catgories.
